src/views/FirstPage.vue

```diff
@@ -166,10 +166,23 @@ export default class FirstPage extends Vue {
 
   async selectWallet(wallet) {
     if(wallet === "ledger") {
-      this.setWalletType("ledger")
-     this.$refs.hardwareWalletConfigRef.show() 
+      
+      try {
+        this.$refs.hardwareWalletConfigRef.show()
+        this.setWalletType("ledger")
+      } catch(err) {
+        this.setShowLoadingSpinner(false)
+        console.log(err)
+      }
     } else if(wallet === "metamask") {
       this.setWalletType("metamask")
+      if(window.ethereum) {
+        window.ethereum.on('accountsChanged', async (accounts) => {
+          if(this.userIsLoggedIn) this.ensureIdentityMappingExists({currentAddress: accounts[0]})
+          this.setCurrentMetamaskAddress(accounts[0])
+        })
```

Comment:
we should also add the code to refresh the page on metamask change from here
https://github.com/loomnetwork/dashboard/pull/271/files#diff-8407d34d5443e5ccd99d1468bc9cb09fL197